src/System.Private.CoreLib/shared/System/Text/Utf16Utility.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal static partial class Utf16Utility
+    {
+        /// <summary>
+        /// Returns true iff the DWORD represents two ASCII UTF-16 characters in machine endianness.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool DWordAllCharsAreAscii(uint value)
+        {
+            return (value & ~0x007F007Fu) == 0;
```

Comment:
There aren't any endianness issues here?

Reply:
Correct - there are no endianness issues here. The reason is that the underlying `char` is already stored as machine-endian.

Consider a string with the UTF-16 code units `[ U+AABB U+CCDD ]`. On a big-endian machine, this will be stored in memory as the bytes 0xAA 0xBB 0xCC 0xDD, and reading this as a uint returns 0xAABBCCDD. On a little-endian machine, this will be stored in memory as the bytes 0xBB 0xAA 0xDD 0xCC, and reading this as a uint returns 0xCCDDAABB. Both of these work with the mask provided in this method.